bpf/bpf_lxc.c

```diff
@@ -1187,6 +1206,9 @@ ipv4_policy(struct __ctx_buff *ctx, int ifindex, __u32 src_label, __u8 *reason,
 			return ret2;
 	}
 
+	if (ct_state.loopback)
```

Comment:
We should add a comment here as well to explain the ingress case.

Reply:
We might also want to compile this condition only in some cases (e.g., when loopback translation is supported) as the compiler is unlikely to be able to optimize it away. For egress, this is not an issue because the compiler should be able to optimize it away already.

Reply:
:+1: yeah in the `ENABLE_HOST_SERVICES_FULL` case (ie socket translation) we won't need this logic.

Reply:
Also if `DISABLE_LOOPBACK_LB` is defined.

Reply:
@pchaigno I already had added a note here, maybe it was lost in the following revisions? Also, we could wrap the condition in the `unlikely` macro since it's in the hot path.

Reply:
> I already had added a note here, maybe it was lost in the following revisions?

Oops yes. I removed it in the second version I tried, when I removed all code changes to the ingress path (I thought the conntrack entry could be enough).

> Also, we could wrap the condition in the unlikely macro since it's in the hot path.

Yep, good call!